### PR TITLE
Add submit type to button

### PR DESCRIPTION
**Pull request by RobinWijnant**
formik relies on a button with type=submit

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -45,10 +47,11 @@ const Button: React.FC<Props> = ({
   });
 
   return (
+    /* eslint-disable-next-line react/button-has-type */
```
Comment (RobinWijnant) [inline src/components/Button/Button.tsx:50]: Sadly i had to disable this rule. This rule does not allow `type` to be set dynamically on `<button/>`.

Comment (bramvanhoutte) [reply src/components/Button/Button.tsx:50]: Hmm, maybe just make the "type" property accessible by the developers?

Comment (RobinWijnant) [reply src/components/Button/Button.tsx:50]: Could do that, although `type='reset'` is considered a bad UX practice. 

> [The Web would be a happier place if virtually all Reset buttons were removed. This button almost never helps users, but often hurts them.](https://ux.stackexchange.com/questions/42771/submit-and-reset-button-whats-the-best-order)

The eslint rule is supposed to stop the developer from changing the `type` at runtime. We don't do this here

Comment (bramvanhoutte) [reply src/components/Button/Button.tsx:50]: Okay, I read the article and this makes sense!! But then I would still make the property `type` which has union type `'submit' | 'button'` and which makes type='reset' impossible

Comment (RobinWijnant) [reply src/components/Button/Button.tsx:50]: Property `type` already exists. What do you propose?
```typescript
interface Props {
  type?: 'primary' | 'secondary';
  ...
}
```

Comment (bramvanhoutte) [reply src/components/Button/Button.tsx:50]: Maybe change the prop currently called `type` to `variant`? And then use `type` for button type

Comment (RobinWijnant) [reply src/components/Button/Button.tsx:50]: There is no difference in meaning between `variant` and `type`. So if `type` has to stay, I started looking for a better alternative for the current `type`. Articles ([1](https://uxplanet.org/primary-secondary-action-buttons-c16df9b36150), [2](https://uxmovement.com/buttons/visual-weight-of-primary-and-secondary-action-buttons/#:~:text=One%20button%20is%20primary%20to,and%20the%20other%20is%20secondary.&text=Visual%20weight%20is%20how%20users,have%20the%20weakest%20visual%20weight.)) describe a `primary` and `secondary` type as a difference in visual weight. Does this make sense to you?
```typescript
interface Props {
  visualWeight?: 'primary' | 'secondary';
  type?: 'button' | 'submit';
}
```
I also came up with:
```typescript
interface Props {
  type?: 'primary' | 'secondary';
  htmlType?: 'button' | 'submit';
}
```
The last one is how `ant-design` resolves this. ([ant-design button](https://ant.design/components/button/#API))

Comment (RobinWijnant) [reply src/components/Button/Button.tsx:50]: I prefer `htmlType`

Comment (bramvanhoutte) [reply src/components/Button/Button.tsx:50]: Yes is good for me!

**Pull request merged**